### PR TITLE
add bt66tech60 handwired

### PR DESCRIPTION
**Pull request by bt66**
add handwired bt66tech keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/bt66tech/bt66tech60/config.h [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xCA04
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER bt66tech
+#define PRODUCT bt66tech 60% 
+#define DESCRIPTION handwired bt66tech60%.
```
Comment (drashna) [inline keyboards/bt66tech/bt66tech60/config.h:26]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/bt66tech/bt66tech60/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    KC_MS_ACCEL1, KC_MS_LEFT,  KC_MS_DOWN,KC_MS_RIGHT, 	 _______,    _______,	 _______,  _______, _______,  _______,    _______, _______,          _______, \
+    KC_MS_ACCEL2,    _______,     _______,    _______,    _______,    _______,    _______,  _______, _______,  _______,              KC_UP,          MO(_FN1), \
+    	 _______,    _______,     _______,                                       _______,                                 KC_LEFT, KC_DOWN, KC_RGHT, _______
+  ),
+
+[_FN1] = LAYOUT_60_ansi(
+    BL_TOGG, BL_STEP,   BL_ON,   BL_OFF,   BL_INC,   BL_DEC,   BL_BRTG,   _______,   _______,   _______,   _______,  BL_DEC,  BL_INC,   _______, \
+     _______, _______,  _______,   _______,  _______,  _______,  _______, _______, _______, _______, _______, _______, _______, _______, \
+     _______,  _______,  _______,  _______,  _______,  _______,  _______, _______, _______, _______, _______, _______,          _______, \
+     _______,  _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______, \
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/keymaps/default/keymap.c:53]:
```suggestion
    KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,                 KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,  KC_EQL,   KC_BSPC,
    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,   		KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,  KC_BSLS,
    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,                 KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,           KC_ENT,
    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,                 	 KC_M,    KC_COMM, KC_DOT,  KC_SLSH,           KC_RSFT,
    KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                          KC_RALT, KC_RGUI, KC_RCTL,  MO(_FN)
  ),

  [_FN] = LAYOUT_60_ansi(
    	  KC_GRV,      KC_F1,       KC_F2,      KC_F3,      KC_F4,      KC_F5,      KC_F6,    KC_F7,   KC_F8,    KC_F9,     KC_F10,  KC_F11,  KC_F12, 	KC_DEL,
    KC_MS_ACCEL0, KC_MS_BTN1,    KC_MS_UP, KC_MS_BTN2, 	 _______,    _______, 	 _______,  _______, _______,  _______, KC_PSCREEN, _______, _______, _______,
    KC_MS_ACCEL1, KC_MS_LEFT,  KC_MS_DOWN,KC_MS_RIGHT, 	 _______,    _______,	 _______,  _______, _______,  _______,    _______, _______,          _______,
    KC_MS_ACCEL2,    _______,     _______,    _______,    _______,    _______,    _______,  _______, _______,  _______,              KC_UP,          MO(_FN1),
    	 _______,    _______,     _______,                                       _______,                                 KC_LEFT, KC_DOWN, KC_RGHT, _______
  ),

[_FN1] = LAYOUT_60_ansi(
    BL_TOGG, BL_STEP,   BL_ON,   BL_OFF,   BL_INC,   BL_DEC,   BL_BRTG,   _______,   _______,   _______,   _______,  BL_DEC,  BL_INC,   _______,
     _______, _______,  _______,   _______,  _______,  _______,  _______, _______, _______, _______, _______, _______, _______, _______,
     _______,  _______,  _______,  _______,  _______,  _______,  _______, _______, _______, _______, _______, _______,          _______,
     _______,  _______,  _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,
```

File: keyboards/bt66tech/bt66tech60/rules.mk [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+MCU_LDSCRIPT = STM32F103x8_stm32duino_bootloader
+BOARD = STM32_F103_STM32DUINO
+
+DFU_ARGS = -d 1eaf:0003 -a2 -R
+DFU_SUFFIX_ARGS = -v 1eaf -p 0003
+
+# project specific files
+#VPATH += keyboards/handwired/bt66tech/bt66tech60/bluepill
+#SRC =	bluepill.c
+
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/rules.mk:15]:
```suggestion
# Bootloader selection
BOOTLOADER = stm32duino
```

File: keyboards/bt66tech/bt66tech60/rules.mk [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	      # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+WS2812_DRIVER = spi
+
+# LAYOUTS = 60_ansi
+
+DEFAULT_FOLDER = handwired/bt66tech/bt66tech60
+
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/rules.mk:30]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
WS2812_DRIVER = spi

DEFAULT_FOLDER = handwired/bt66tech/bt66tech60
```

File: keyboards/bt66tech/bt66tech60/readme.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# bt66tech60
+
+![bt66tech60](https://i.imgur.com/8eJ4liU.jpg)
+
+*A short description of the keyboard/project*
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/readme.md:5]: Please fill this out.

File: keyboards/bt66tech/bt66tech60/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    _FN,
+    _FN1
+};
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    QMKBEST = SAFE_RANGE,
+    QMKURL
+};
+
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/keymaps/default/keymap.c:30]:
```suggestion
```

File: keyboards/bt66tech/bt66tech60/bt66tech60.c [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+#define LED_OFF()
+#define LED_TGL()
+#endif
+
+void matrix_init_kb(void){
+    LED_ON();
+    wait_ms(500);
+    LED_OFF();
+    matrix_init_user();
+}
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/bt66tech60.c:38]:
```suggestion
```

File: keyboards/bt66tech/bt66tech60/config.h [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xCA04
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER bt66tech
+#define PRODUCT bt66tech 60% 
+#define DESCRIPTION A handwired 60% keyboard.
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/config.h:26]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/bt66tech/bt66tech60/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+            if (record->event.pressed) {
+                // when keycode QMKURL is pressed
+                SEND_STRING("https://qmk.fm/\n");
+            } else {
+                // when keycode QMKURL is released
+            }
+            break;
+    }
+    return true;
+}
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/keymaps/default/keymap.c:78]:
```suggestion
```

File: keyboards/bt66tech/bt66tech60/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+    _BASE,
+    _FN,
+    _FN1
+};
+
+
+
+
+
+
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/keymaps/default/keymap.c:29]:
```suggestion
```

File: keyboards/bt66tech/bt66tech60/readme.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
+# bt66tech60
+
+![bt66tech60](https://i.imgur.com/8eJ4liU.jpg)
+
+*a handwired 60% keyboard with stm32t*
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/readme.md:5]:
```suggestion
A handwired 60% keyboard with stm32
```

File: keyboards/bt66tech/bt66tech60/readme.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+
+* Keyboard Maintainer: [bt66](https://github.com/bt66)
+* Hardware Supported: *handwired, STM32*
+* Hardware Availability: *anywhere*
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/readme.md:9]:
```suggestion
* Hardware Supported: handwired, STM32
* Hardware Availability: anywhere
```

File: keyboards/bt66tech/bt66tech60/rules.mk [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+BOOTLOADER = stm32duino
+
+
+
+
+
+
+
+
+
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/rules.mk:14]:
```suggestion
```

File: keyboards/bt66tech/bt66tech60/rules.mk [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+WS2812_DRIVER = spi
+DEFAULT_FOLDER = handwired/bt66tech/bt66tech60
+
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/rules.mk:34]:
```suggestion

DEFAULT_FOLDER = handwired/bt66tech/bt66tech60
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/bt66tech/bt66tech60/info.json [hunk 15]
```diff
@@ -0,0 +1,15 @@
+{
+    "keyboard_name": "bt66tech60",
+    "url": "",
+    "maintainer": "bt66tech",
+    "width": 15,
+    "height": 5,
+    "layouts": {
+        "LAYOUT": {
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/info.json:8]:
```suggestion
        "LAYOUT_60_ansi": {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/bt66tech/bt66tech60/rules.mk (hunk 14 above)
Comment (bt66) [reply keyboards/bt66tech/bt66tech60/rules.mk:26]: add LAYOUTS = 60_ansi in rules.mk make my pull request check error , 
are you sure about this?
if you sure can you tell me how fix git check error after add LAYOUTS = 60_ansi

Comment (fauxpark) [reply keyboards/bt66tech/bt66tech60/rules.mk:26]: Your layout macro and info.json look correct, so most likely it is just some community layout failing to build. You can safely ignore it.

**Review by fauxpark — CHANGES_REQUESTED** [review]
.

File: keyboards/bt66tech/bt66tech60/rules.mk [hunk 16]
```diff
@@ -0,0 +1,30 @@
+# MCU name
+MCU = STM32F103
+
+# Bootloader selection
+BOOTLOADER = stm32duino
+
+# Build Options
+#   change yes to no to disable
+
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/rules.mk:9]:
```suggestion
#
```

File: keyboards/bt66tech/bt66tech60/rules.mk [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+WS2812_DRIVER = spi
+
+DEFAULT_FOLDER = handwired/bt66tech/bt66tech60
```
Comment (fauxpark) [inline keyboards/bt66tech/bt66tech60/rules.mk:25]:
```suggestion
DEFAULT_FOLDER = bt66tech/bt66tech60
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**